Libs/Reflex/src/test/java/reflex/MatchTest.java

```diff
@@ -0,0 +1,284 @@
+package reflex;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Map;
+
+import org.antlr.runtime.ANTLRStringStream;
+import org.antlr.runtime.CommonTokenStream;
+import org.antlr.runtime.RecognitionException;
+import org.antlr.runtime.tree.CommonTree;
+import org.antlr.runtime.tree.CommonTreeNodeStream;
+import org.apache.log4j.AppenderSkeleton;
+import org.apache.log4j.Logger;
+import org.apache.log4j.spi.LoggingEvent;
+import org.junit.Test;
+
+import rapture.common.api.ScriptingApi;
+import reflex.node.MatchNode;
+import reflex.node.ReflexNode;
+import reflex.util.InstrumentDebugger;
+import reflex.value.ReflexValue;
+import reflex.value.internal.ReflexNullValue;
+
+public class MatchTest {
+	private static Logger log = Logger.getLogger(MatchTest.class);
+
+	@Test
+	public void happyPathMatch() throws RecognitionException {
+		String program = "ident = 5;\n" + "match ident do\n" + "is > 4 do\n" + "println('Success');\n"
+				+ "end\n" + "is < 6 do\n" + "println('Fail');\n" + "end\n" + "otherwise do\n" + "println('Fail');\n"
+				+ "end\n" + "end";
+
+		String output = runScript(program, null);
+		assertEquals("Success", output);
+	}
+	
+	@Test
+	public void testAsClause() throws RecognitionException {
```

Comment:
There's a better way of writing these tests, where the program is in the resources and you refer to them from that... (see most of the other tests!!!)